src/ray/rpc/worker/core_worker_client.h

```diff
@@ -419,47 +363,7 @@ class CoreWorkerClient : public std::enable_shared_from_this<CoreWorkerClient>,
   /// The client will guarantee no more than kMaxBytesInFlight bytes of RPCs are being
   /// sent at once. This prevents the server scheduling queue from being overwhelmed.
   /// See direct_actor.proto for a description of the ordering protocol.
-  void SendRequests() {
-    absl::MutexLock lock(&mutex_);
-    auto this_ptr = this->shared_from_this();
-
-    while (!send_queue_.empty() && rpc_bytes_in_flight_ < kMaxBytesInFlight) {
-      auto pair = std::move(*send_queue_.begin());
-      send_queue_.pop_front();
-
-      auto request = std::move(pair.first);
-      int64_t task_size = RequestSizeInBytes(*request);
-      int64_t seq_no = request->sequence_number();
-      request->set_client_processed_up_to(max_finished_seq_no_);
-      rpc_bytes_in_flight_ += task_size;
-
-      auto rpc_callback =
-          [this, this_ptr, seq_no, task_size, callback = std::move(pair.second)](
-              Status status, rpc::PushTaskReply &&reply) {
-            {
-              absl::MutexLock lock(&mutex_);
-              if (seq_no > max_finished_seq_no_) {
-                max_finished_seq_no_ = seq_no;
-              }
-              rpc_bytes_in_flight_ -= task_size;
-              RAY_CHECK(rpc_bytes_in_flight_ >= 0);
-            }
-            SendRequests();
-            callback(status, std::move(reply));
-          };
-
-      RAY_UNUSED(INVOKE_RPC_CALL(CoreWorkerService,
-                                 PushTask,
-                                 *request,
-                                 std::move(rpc_callback),
-                                 grpc_client_,
-                                 /*method_timeout_ms*/ -1));
-    }
-
-    if (!send_queue_.empty()) {
-      RAY_LOG(DEBUG) << "client send queue size " << send_queue_.size();
-    }
-  }
+  void SendRequests();
```

Comment:
The compilation speed could be further improve with forward declaration (i.e. L384)
It's likely worth it because grpc header file is included by it.